src/VkLayer_GF_amber_scoop/src/vulkan_formats.cc

```diff
@@ -0,0 +1,418 @@
+// Copyright 2020 The gf-layers Project Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "VkLayer_GF_amber_scoop/vulkan_formats.h"
+
+// <cstdio> and <cstdlib> required by LOG and ASSERT macros, but IWYU doesn't
+// notice it, so we need to manually keep the includes.
+#include <cstdio>   // IWYU pragma: keep
+#include <cstdlib>  // IWYU pragma: keep
+
+#include "gf_layers_layer_util/logging.h"
+
+namespace gf_layers::amber_scoop_layer {
+
+const char* VkFormatToAmberFormatName(VkFormat vk_format) {
+  if (vk_format == VK_FORMAT_A1R5G5B5_UNORM_PACK16) {
+    return "A1R5G5B5_UNORM_PACK16";
+  }
+  if (vk_format == VK_FORMAT_A2B10G10R10_SINT_PACK32) {
+    return "A2B10G10R10_SINT_PACK32";
+  }
```

Comment:
Could we use a switch here?

Reply:
Yeah, definitely we could. I think the reaseon I used ifs is that clang-tidy requires all enum values to be listed, not just the ones Amber supports, ie. we cannot use `default:` case. I'll check if the warning/error can be disabled with a pragma or something.